### PR TITLE
Elg 263

### PR DESCRIPTION
**Pull request by gcleary**
Add bitmap field to carry 'Used AP' info from embedded-server to client.

### CODE REVIEW

File: el.proto [hunk 1]
```diff
@@ -152,6 +152,7 @@ message Rs {
     uint32 hpe = 3;
     Source source = 4;
     ClientConfig config = 5;
+    bytes used_aps = 6;
```
Comment (tboinske-skyhook) [inline el.proto:155]: This should include a comment which defines the meaning of this field. It should indicate that it's a bit-vector, and define the bit ordering (i.e., the meaning of the least significant bit).

Comment (gcleary) [reply el.proto]: Yup. Done

File: el.proto [hunk 2]
```diff
@@ -152,9 +152,13 @@ message Rs {
     uint32 hpe = 3;
     Source source = 4;
     ClientConfig config = 5;
+    bytes used_aps = 6;               // Bitmap denoting whether an AP was Used by server
+                                      // Big endian. Length = number of (APs + 7) / 8
+                                      // LSB of last byte corresponds to first AP
+                                      // LSB of second to last byte corresponds to ninth AP etc
```
Comment (tboinske-skyhook) [inline el.proto:158]: This looks good, but you should also mention what the assumed AP ordering is (the order of the APs as they are specified in the request, I believe).

Comment (gcleary) [reply el.proto]: Yup. Done

File: el.proto [hunk 3]
```diff
@@ -70,6 +70,7 @@ message Aps {
     repeated int32 frequency = 5;    // Not valued if all frequency are the same.
     repeated uint32 neg_rssi = 6;    // -rssi
     repeated uint32 age = 7;         // Not valued if all ages are the same.
+    repeated uint32 vap_delta = 8;   // AP index(8bit), nibble index(4bit), replacement nibble(4bit)
```
Comment (tboinske-skyhook) [inline el.proto:73]: This comment could use some fleshing. It accounts for only 8+4+4 = 16 bits, but the field is a 32-bit array. So perhaps something like:
```c
// An array of 16-bit unsigned integers, with each integer broken down (left to right) in the following way: 
// AP index(8bit), nibble index(4bit), replacement (patch) nibble(4bit)
```

Comment (tboinske-skyhook) [reply el.proto:73]: However, a perhaps bigger issue: The other "arrays" in this message all have a length which corresponds to the number of (non-virtual) APs. Except for this new entry, whose length is just the total number of "child" virtual APs. This isn't a practical problem (it'll work), but it is at least a slight conceptual problem. Would it make sense to move `vap_delta` into the `Rq` message, or even into its own "VirtualAps" message?

Comment (gcleary) [reply el.proto:73]: Agree with comment suggestion.

I struggled with where to put the Virtual AP members. I came down on the side that they are additional APs and should therefore be part of that message. The server adds them to the ap_scans list. We can improve the name to imply it's different use? e.g. vap_members.

Comment (tboinske-skyhook) [reply el.proto:73]: Given that its presence in the Aps message "breaks" the overall semantics of the message, I would prefer moving it out, assuming doing so doesn't cause other problems.

**Comment by gcleary** [comment]
abandon this in deference to branch ELG-273

**Pull request closed without merging**